Source/Charts/Data/Implementations/Standard/ChartData.swift

```diff
@@ -447,57 +270,51 @@ open class ChartData: NSObject, ExpressibleByArrayLiteral
     }
     
     /// - returns: The DataSet that contains the provided Entry, or null, if no DataSet contains this entry.
-    @objc open func getDataSetForEntry(_ e: ChartDataEntry!) -> ChartDataSetProtocol?
+    @objc open func getDataSetForEntry(_ e: ChartDataEntry) -> Element?
     {
-        guard e != nil else { return nil }
-
         return first { $0.entryForXValue(e.x, closestToY: e.y) === e }
     }
 
     /// - returns: The index of the provided DataSet in the DataSet array of this data object, or -1 if it does not exist.
-    @objc open func indexOfDataSet(_ dataSet: ChartDataSetProtocol) -> Int
+    @objc open func index(of dataSet: Element) -> Index
     {
         return index(where: { $0 === dataSet }) ?? -1
     }
     
     /// - returns: The first DataSet from the datasets-array that has it's dependency on the left axis. Returns null if no DataSet with left dependency could be found.
-    @objc open func getFirstLeft(dataSets: [ChartDataSetProtocol]) -> ChartDataSetProtocol?
+    @objc open func getFirstLeft(dataSets: [Element]) -> Element?
     {
         return first { $0.axisDependency == .left }
     }
     
     /// - returns: The first DataSet from the datasets-array that has it's dependency on the right axis. Returns null if no DataSet with right dependency could be found.
-    @objc open func getFirstRight(dataSets: [ChartDataSetProtocol]) -> ChartDataSetProtocol?
+    @objc open func getFirstRight(dataSets: [Element]) -> Element?
     {
         return first { $0.axisDependency == .right }
     }
     
     /// - returns: All colors used across all DataSet objects this object represents.
-    // TODO: This should return a non-optional array
-    @objc open func getColors() -> [NSUIColor]?
+    @objc open func getColors() -> [NSUIColor]
```

Comment:
seems this func is never being used? I'm ok to keep it.

Reply:
Agreed. I'm guessing it only exists as a convenience method for consumers.